nuxeo-features/nuxeo-platform-rendition/nuxeo-platform-rendition-core/src/main/java/org/nuxeo/ecm/platform/rendition/service/RenditionServiceImpl.java

```diff
@@ -239,6 +259,8 @@ public void registerContribution(Object contribution, String extensionPoint, Com
             renditionDefinitionProviderRegistry.addContribution((RenditionDefinitionProviderDescriptor) contribution);
         } else if (STORED_RENDITION_MANAGERS_EP.equals(extensionPoint)) {
             storedRenditionManagerDescriptors.add(((StoredRenditionManagerDescriptor) contribution));
+        } else if (DEFAULT_RENDITION_EP.equals(extensionPoint)) {
+            defaultRenditionDescriptor = ((DefaultRenditionDescriptor) contribution);
```

Comment:
You need to also take into account descriptor unregistration. So you need to keep a list of contributions, and of the last one. And in unregister remove from the list and re-set the defaultRenditionDescriptor to the last one. See for example `redirectResolverContributions` / `redirectResolver` in `DownloadServiceImpl`.

Reply:
:+1: